proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +111,336 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
+[Matrix Transpose]: #matrix-transpose
+[Minimum Support Set]: #minimum-support-set
+[CheckFeatureSupport]: #check-feature-support
 
 
-**Reduce Sum Accumulate**
+### Type Interpretations
 
-Accumulates the components of a vector atomically (with device scope) to the corresponding elements of an array in
-memory.
-
-```
-template<typename T, uint M>
-void ReduceSumAccumulate(vector<T, M> InputVector,
-                         RWByteAddressBuffer Buf,
-                         uint BufOffset);
+The various "interpretation" arguments specify a value from the following enum:
 
+```c++
+enum class DXILTypeInterpretation :uint {
+  Float16               = 0,
+  Float32               = 1,
+  UnsignedInt8          = 2,
+  UnsignedInt16         = 3,
+  UnsignedInt32         = 4,
+  SignedInt8            = 5,
+  SignedInt16           = 6,
+  SignedInt32           = 7,
+  SignedInt8x4Packed    = 8,
+  UnsignedInt8x4Packed  = 9,
+  FloatE4M3             = 10,
+  FloatE5M2             = 11,
+  Unsupported           = 32
+};
 ```
 
-*InputVector* is an M component vector of type T.
+For matrices and vectors that are specified by resource handles and stored in
+raw-buffers, the interpretation value directly specifies the element type.  It
+is invalid to specify a packed interpretation in these cases.
 
-*Buf* is the array into which the *InputVector* is accummulated. The base address and *BufOffset* of the buffer
- must be 64B aligned.
+For input vectors that come from variables there is a distinction between the
+physical type and the logical type. The **input interpretation** argument for
+these vectors describes how to convert from the physical to logical type. This
+allows elements to be interpreted as types not natively supported by HLSL, e.g.
+uint8/sint8. For packed interpretations, a single physical element can expand
+into multiple logical elements.
 
-*BufOffset* is the offset to the first element of the array to which the *InputVector* is accummulated. It is 64B aligned.
+[CheckFeatureSupport] can be used to determine what combinations of **TYi**,
+**input interpretation**, **matrix interpretation**, **matrix transpose**,
+**bias vector interpretation** and **TYo** are supported on a particular
+implementation. A list of combinations that are guaranteed to be supported on
+all implementations can be found in [Minimum Support Set]. Note that there is no
+guaranteed support for **matrix tranpose**, and so it must always queried. s
+queried.
 
-The device should be queried in [Check Feature Support](#check-feature-support) to determine datatypes of InputVector supported along
-with the AccumulationType. An exception to this rule is the set of combinations guaranteed to be supported on all
-devices supporting this feature. See [Minimum Support Set](#minimum-support-set).
-
-### Example HLSL Shader
-
-// XXX TODO
+Non-"Packed" type interpretations are used to request arithmetic conversions. Input type must be a 32-bit or 16-bit
```

Comment:
Perhaps the conversion rules should have a separate sections called "Conversion Rules" to draw attention to them. The way it is laid out right now, it might be easiy missed by a reader.

Reply:
> Perhaps the conversion rules should have a separate sections called "Conversion Rules" to draw attention to them.

How does this look?  Added it as a sub-heading.